### PR TITLE
Azue admin add update user

### PR DESCRIPTION
**Pull request by jdf400-rapid7**
### Description

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR.
4. Add UI screenshot of the workflow used for testing
5. Add UI screenshot of the job output used for testing
6. Add UI screenshot of the artifact (See rules in UI Checklist) used for testing

### CODE REVIEW

**Comment by jdf400-rapid7** [comment]
⇒  icon-plugin run -R tests/upate_user_info.json -A
INFO[0000] Running command:  docker run --rm -i rapid7/azure_ad_admin:1.5.0 run < tests/upate_user_info.json 
{"body": {"log": "Connect: Connecting...\nUpdating auth token...\nGetting token from: https://login.microsoftonline.com/5c824599-dc8c-4d31-96fb-3b886d4f8f10/oauth2/token\nAuthentication request status: 200\nAuthentication Token: ****************4JpGg\nrapid7/Azure AD Admin:1.5.0. Step name: update_user_info\nUpdating info for user: jschipp@komanddev.onmicrosoft.com\n", "status": "ok", "meta": {}, "output": {"success": true}}, "version": "v1", "type": "action_event"}
## Assessment
### Run
<details>
```
{
  "body": {
    "log": "Connect: Connecting...\nUpdating auth token...\nGetting token from: https://login.microsoftonline.com/5c824599-dc8c-4d31-96fb-3b886d4f8f10/oauth2/token\nAuthentication request status: 200\nAuthentication Token: ****************4JpGg\nrapid7/Azure AD Admin:1.5.0. Step name: update_user_info\nUpdating info for user: jschipp@komanddev.onmicrosoft.com\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}
```
<summary>
docker run --rm -i rapid7/azure_ad_admin:1.5.0 run < tests/upate_user_info.json
</summary>
</details>
### UI
Screenshots of the plugin being used in Komand for validating use go below this line.
#### Workflow Builder
#### Job
#### Artifact

**Comment by jschipp-r7** [comment]
@jdf400-rapid7 Can you post validator output?

File: azure_ad_admin/plugin.spec.yaml [hunk 1]
```diff
@@ -471,6 +471,51 @@ actions:
...
         required: false
+  update_user_info:
+    title: Update User Information
+    description: Update a users information
+    input:
+      city:
+        title: City
+        type: string
+        description: The city in which the user is located
+        required: false
```
Comment (jschipp-r7) [inline azure_ad_admin/plugin.spec.yaml:482]: Can you add an example key for each of these inputs with valid/working examples where possible and then regenerate on icon-plugin 4.8.4.

```suggestion
        required: false
        example: Boston
```

File: azure_ad_admin/plugin.spec.yaml [hunk 2]
```diff
@@ -387,7 +387,7 @@ actions:
         description: Body of the email to be sent out. Use $password to place the generated password
       notify_recipient:
         required: true
-        title: Recipient of creation email
+        title: Recipient of Creation Email
```
Comment (jschipp-r7) [inline azure_ad_admin/plugin.spec.yaml:390]: Requires a major version bump because the variable change will break any workflows

**Comment by jschipp-r7** [comment]
```
$ make validate .
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Colors.mk ../tools/Makefiles/Helpers.mk
--
[*] Running validators
You are using pip version 18.1, however version 20.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 261.728ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] Validating markdown...
[SUCCESS] Passes markdown linting
```

**Pull request merged**